timesketch/lib/analyzers/geoip.py

```diff
@@ -0,0 +1,311 @@
+        """
+        try:
+            response = self.city(ip_address)
+        except geoip2.errors.AddressNotFoundError as exception:
+            logging.debug('IP address {0} not found.'.format(
+                ip_address))
+            return None
+        except Exception as error:
+            logging.error('Error while geolocating {0} - {1}'.format(
+                ip_address, error))
+            return None
+
+        latitude = response.location.latitude
+        longitude = response.location.longitude
+        country_name = response.country.name
+        iso_code = response.country.iso_code
+        city = response.city.name
+
+        return (iso_code, latitude, longitude, country_name, city)
+
+
+class MaxMindGeoWebClient(geoip2.webservice.Client, GeoIpClientAdapter):
+    """A GeoIP client using the MaxMind web service api."""
+
+    def __init__(self):
+        self._account_id = current_app.config.get('MAXMIND_WEB_ACCOUNT_ID')
+        self._license_key = current_app.config.get('MAXMIND_WEB_LICENSE_KEY')
+        self._host = current_app.config.get('MAXMIND_WEB_HOST')
+        if not self._account_id:
+            raise RuntimeError('MaxMind Account ID not set.')
+        if not self._license_key:
+            raise RuntimeError('MaxMind License key not set.')
+        if not self._host:
+            raise RuntimeError('MaxMind host not set.')
+        super().__init__(self._account_id, self._license_key, host=self._host)
+
+    def ip2geo(self, ip_address) -> Union[Tuple[str, str, str, str], None]:
+        """Perform a IP to geolocation lookup.
+
+        Args:
+            ip_address - the IPv4 or IPv6 address to geolocate
+
+        Returns:
+            Either:
+            A tuple comprising of the following in order
+            - iso_code (str) - the ISO 31661 alpha-2 code of the country
+            - latitude (str) - the north-south coordinate
+            - longitude (str) - the east-west coordinate
+            - country_name (str) - the full country name
+            - city (str) - the city name that approximates the location
+            Or None:
+            - when the IP address does not have a resolvable location
+        """
+        try:
+            response = self.city(ip_address)
+        except geoip2.errors.AddressNotFoundError as exception:
+            logging.debug('IP address {0} not found.'.format(
+                ip_address))
+            return None
+        except Exception as error:
+            logging.error('Error while geolocating {0} - {1}'.format(
+                ip_address, error))
+            return None
+
+        latitude = response.location.latitude
+        longitude = response.location.longitude
+        country_name = response.country.name
+        iso_code = response.country.iso_code
+        city = response.city.name
+
+        return (iso_code, latitude, longitude, country_name, city)
+
+
+class BaseGeoIpSketchPlugin(interface.BaseAnalyzer):
```

Comment:
I know this naming comes from the scaffolder, but we are starting to rename because we have deprecated the notion of SketchAnalyzer vs IndexAnalyzer. Now we only have Analyzer :)

```suggestion
class BaseGeoIpAnalyzer(interface.BaseAnalyzer):
```